File: java-code/src/Solns/squares_of_a_sorted_array_977.java
package Solns;

public class squares_of_a_sorted_array_977 {

    public static int[] sortedSquares(int[] nums) {
		int left = 0;
		int right = nums.length - 1;
		
		int[] toReturn = new int[nums.length];
		
		int idx = nums.length - 1;
		
		while(left <= right) {
			if(Math.pow(nums[left], 2) < Math.pow(nums[right], 2)) {
				toReturn[idx] = (int)Math.pow(nums[right], 2);
				right--;
			}
			else {
				toReturn[idx] = (int)Math.pow(nums[left], 2);
				left++;
			}
			idx--;
		}
		return toReturn;
    }
}
	
	public static void main(String[] args) {
		int[] a = sortedSquares(new int[] {-7,-3,2,3,11});
		for(int i = 0; i < a.length; i++)
			System.out.println(a[i]);
	}
}

File: python-code/Two_Pointers/squares_of_a_sorted_array_977.py
def sortedSquares(nums):
#   If list is empty
    if len(nums) == 0:
        return []

    left = 0
    right = len(nums) - 1
    newNums = []

    while right >= left:
        print('loop')
        if abs(nums[right]) > abs(nums[left]):
            newNums.insert(0, nums[right]**2)
            right -= 1
        else:
            newNums.insert(0, nums[left]**2)
            left += 1

    print(newNums)


def main():
    a = [0, 1, 9, 16, 100]

    # print(sortedSquares(a))

    b = [-7, -3, -2, -1]
    # print(sortedSquares(b))

    c = [-10, -8, -7, -2, 0, 1, 3, 4, 7, 9]
    # sortedSquares(c)

    d = [-4,-1,0,3,10]
    sortedSquares(d)

main()